src/Interpreters/ClusterDiscovery.cpp

```diff
@@ -124,21 +125,33 @@ ClusterDiscovery::ClusterDiscovery(
 
     for (const auto & key : config_keys)
     {
-        String prefix = config_prefix + "." + key + ".discovery";
-        if (!config.has(prefix))
+        String cluster_config_prefix = config_prefix + "." + key + ".discovery";
+        if (!config.has(cluster_config_prefix))
             continue;
 
+        String zk_root = config.getString(cluster_config_prefix + ".path");
+        if (zk_root.empty())
+            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "ZooKeeper path for cluster '{}' is empty", key);
+
+        const auto & password = config.getString(cluster_config_prefix + ".password", "");
+        const auto & cluster_secret = config.getString(cluster_config_prefix + ".secret", "");
+        if (!password.empty() && !cluster_secret.empty())
+            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "Mutually exclusive options 'password' and 'secret' are specified for cluster '{}'", key);
```

Comment:
Let's make it more human friendly
```suggestion
            throw Exception(ErrorCodes::NO_ELEMENTS_IN_CONFIG, "Both 'password' and 'secret' are specified for cluster '{}', only one option can be used at the same time", key);
```